documentation/guide-how-to-create-a-devon4quarkus-app.asciidoc

```diff
@@ -0,0 +1,213 @@
+
+More Information on this Chapter{Chapter  Quarkus Testing ?} or this Guide https://quarkus.io/guides/dev-ui
+
+=== Testing the application 
+When in developer mode (via mvn quarkus:dev), Quarkus can automatically and continuously run your unit tests. Quarkus supports Junit 5 tests and also has generated a simple test for us. When the devon4quarkus application is running you should see in your command prompt:
+----
+Tests paused, press [r] to resume, [w] to open the browser, [h] for more options> 
+----
+The test is pretty simple and just checks if "Hello RESTEasy" will be returned. We can edit the *getting-started/src/java/org/acme/getting/started/GreetingResource.java* while the application is still running to:
+
+[source, java]
+----
+@Path("/hello")
+public class GreetingResource {
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String hello() {
+        return "Hello RESTHard";
+    }
+}
+----
+
+and by rerunning the test with `r`, the test should fail now. You can also test the application before running it with `mvn test`. Reset the return string to *"Hello RestEasy"* and we can continue.
+More Information on this Chapter{Chapter Testing ?} 
+
+=== Packaging the application
+First, let's package the application with `mvn package` and produce:
+
+
+. *target/getting-started-1.0.0-SNAPSHOT.jar* +
+containing just the classes and resources of the projects, it’s the regular artifact produced by the Maven build
+. *target/quarkus-app/quarkus-run.jar* +
+being an executable jar. Be aware that it’s not an über-jar as the dependencies are copied into several subdirectories (and would need to be included in any layered container image).
+
+You can run the packaged application with this command `java -jar target/quarkus-app/quarkus-run.jar` and check http://localhost/hello.
+
+
+Let’s now produce a native executable for our application. It improves the startup time of the application and produces a minimal disk footprint. The executable would have everything to run the application including the "JVM" (shrunk to be just enough to run the application), and the application.
+
+
+Building a native executable requires using a distribution of GraalVM and a configured $GRAALVM_HOME. You can create a native application without GraalVM and use a multi-stage Docker build to run Maven inside a Docker container that embeds GraalVM. This will be explained in this chapter <<Build native executables without GraalVM>> or this https://quarkus.io/guides/building-native-image#container-runtime[guide]
+
+
+The usage of GraalVM is easier on Linux, for Windows the setup for the environment takes some more steps.
+
+.Operating system
+
+. Linux
+* Download the *Community Edition of GraalVM* from https://github.com/graalvm/graalvm-ce-builds/releases[here] and unpack it like you would any other archive. Make sure to download and install at Java 11 version. 
```

Comment:
I think we should not mention this here as it is not part of the "How to create an application". Either we should add a separate guide for setting up GraalVM or just add a link to an existing guide.

Reply:
You either need an installed container enviroment or GraalVM to build a nativ executable and i think this should also be part of the guide. Maybe change the structure of the guide so that the first way will be to build a native executable with Docker and after that the GraalVM part so they can skip it easily.

Reply:
Ok, but in my opinion we don't need the part about installation in this detail. I would just refer to the official Quarkus guide https://quarkus.io/guides/building-native-image

Reply:
I removed the part.